pkg/apis/pipeline/v1/taskrun_validation_test.go

```diff
@@ -142,6 +184,32 @@ func TestTaskRun_Validate(t *testing.T) {
 			},
 		},
 		wc: config.EnableAlphaAPIFields,
+	}, {
+		name: "propagating partial object params with different provided and default names",
```

Comment:
I think I left a similar comment on a previous PR but params don't have default names (just default values), so I think this is a bit confusing-- it looks like here the task spec is declaring another param task-words-2 and not using it (which I'm sort of surprised is valid); the test case name should reflect that. Also what's a "partial" param?

Reply:
Yes, I had that for increasing coverage. I will use it to avoid confusion. Partial params in my mind (which you appropriately worded in the next comment below) is really params where some of the keys are provided by the task run/pipelinerun and others are defaulted in the task/pipeline spec. (i.e. a case where only some of the params are explicitly provided).

Reply:
I updated the names. Should be ok now.